proxy/src/source.rs

```diff
@@ -171,50 +172,49 @@ where
         id
     }
 
-    fn get_all_projects(&self) -> Vec<Project> {
-        // TODO(xla): Return proper error.
-        self.registry_client
+    fn get_all_projects(&self) -> Result<Vec<Project>, RegistryError> {
+         Ok(self
+            .registry_client
             .list_projects()
-            .wait()
-            .expect("osc client list projects failed")
+            .wait()? // bubble error up if registry client throws anything
             .into_iter()
             .take(10)
             .flat_map(|id| {
-                let maybe_project = self
-                    .registry_client
-                    .get_project(id)
-                    .wait()
-                    .expect("get_project failed");
-
-                match maybe_project {
-                    Some(project) => Some(self.enrich_members(Project::from(project))),
-                    None => None,
+                match self.get_project(id.into()) {
+                    Ok(maybe_project) => match maybe_project {
+                        Some(project) => Ok(project),
+                        None => {
+                            // TODO possible data consistency issue. retry?
+                            panic!("Project listed by registry but could not be found.")
+                        },
+                    },
+                    Err(err) => Err(err),
                 }
             })
-            .collect()
+            .collect())
     }
 
-    fn get_project(&self, id: ProjectId) -> Option<Project> {
+    fn get_project(&self, id: ProjectId) -> Result<Option<Project>, RegistryError> {
         let maybe_project = self
             .registry_client
-            .get_project(id.into())
-            .wait()
-            .expect("get project failed");
+            .get_project(id.try_into().unwrap())
+            .wait()?;
 
-        match maybe_project {
+        let result = match maybe_project {
```

Comment:
This can be simplified to `maybe_project.map(|p| self.enrich_members(Project::from(p)))`